addons/purchase_stock/models/stock_rule.py

```diff
@@ -42,6 +43,10 @@ def _run_buy(self, procurements):
         procurements_by_po_domain = defaultdict(list)
         for procurement, rule in procurements:
 
+            if float_compare(procurement.product_qty, 0, precision_rounding=procurement.product_uom.rounding) < 0:
```

Comment:
from https://github.com/odoo/odoo/commit/a67d6eff4543e015b0cc99637a7281eeb5d5cd46 you can use :

`fields.Float.compare(....)`

indeed

```
from odoo.tools import float_compare
float_compare()
```